flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestQueue.java

```diff
@@ -181,19 +179,18 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object msg) throws Exc
 				return;
 			}
 
-			// Cancel the request for the input channel
+			// remove reader from queue of available readers
 			int size = availableReaders.size();
 			for (int i = 0; i < size; i++) {
 				NetworkSequenceViewReader reader = pollAvailableReader();
-				if (reader.getReceiverId().equals(toCancel)) {
-					reader.releaseAllResources();
-					markAsReleased(reader.getReceiverId());
-				} else {
+				if (reader != null && !reader.getReceiverId().equals(toCancel)) {
 					registerAvailableReader(reader);
 				}
 			}
 
-			allReaders.remove(toCancel);
+			// remove reader from queue of all readers and release its resource
+			NetworkSequenceViewReader toRelease = allReaders.remove(toCancel);
```

Comment:
The `toRelease` could be null if it was already released on server side (any network exceptions) before receiving cancel message from client side.

I think it might be proper to ignore non-existing readers here. And it also makes sense to remove `PartitionRequestQueue#released`